operator/operator.make

```diff
@@ -20,32 +15,18 @@ _work/bin/operator-sdk-$(OPERATOR_SDK_VERSION):
 operator-generate-k8s: controller-gen
 	GOROOT=$(shell $(GO) env GOROOT) $(CONTROLLER_GEN) object paths=./pkg/apis/...
 
-# find or download if necessary controller-gen
-# this make target is copied from Makefile generated
-# by operator-sdk init
-controller-gen:
-ifeq (, $(shell which controller-gen))
-	@{ \
-	set -e; \
-	CONTROLLER_GEN_TMP_DIR=$$(mktemp -d) ;\
-	cd $$CONTROLLER_GEN_TMP_DIR ;\
-	GOPATH=$$($(GO) env GOPATH) ;\
-	$(GO) mod init tmp ;\
-	$(GO) get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.3.0 ;\
-	rm -rf $$CONTROLLER_GEN_TMP_DIR; \
-	}
-CONTROLLER_GEN=$(GOPATH)/bin/controller-gen
-else
-CONTROLLER_GEN=$(shell which controller-gen)
-endif
+# Build controller-gen from source.
+_work/bin/controller-gen-$(CONTROLLER_GEN_VERSION):
+	mkdir -p $(@D)
+	GOBIN=$(abspath $(@D)) $(GO) install sigs.k8s.io/controller-tools/cmd/controller-gen@$(CONTROLLER_GEN_VERSION)
+	ln -sf controller-gen $@
 
-ifeq ($(REPO_ROOT), )
-REPO_ROOT=$(shell pwd)
-endif
-MANIFESTS_DIR=$(REPO_ROOT)/deploy/kustomize/olm-catalog
-CATALOG_DIR=$(REPO_ROOT)/deploy/olm-catalog
-BUNDLE_DIR=$(REPO_ROOT)/deploy/olm-bundle/$(MAJOR_MINOR_PATCH_VERSION)
-CRD_DIR=$(REPO_ROOT)/deploy/crd
+CONTROLLER_GEN=_work/bin/controller-gen-$(CONTROLLER_GEN_VERSION)
```

Comment:
I don't see a reason for using absolute paths. Doing everything relative to where we run make is easier.